### ISSUE TITLE
Do you plan to bring this to F-Droid?

### ISSUE THREAD

**Issue by katrinleinweber** [NiciDieNase/chaosflix#18]
Google-avoiding users would surely appreciate this. https://f-droid.org/en/docs/Submitting_to_F-Droid_Quick_Start_Guide/ seems to provide the basic info about their process. In any case: 1k thanks for this app :-)

**Comment by NiciDieNase** [comment]
That's definitely on my Todo list. I might look into it after Congress.

**Comment by theScrabi** [comment]
If not yet done I'll do it tomorrow.

**Comment by theScrabi** [comment]
So I wrote the [necessary meta file for fdroid](https://gitlab.com/derSchabi/fdroiddata/commit/25dde774b2def38916083e7492581a1cd43d4949). However when I try to build with `fdroid build de.nicidienase.chaosflix` it I get this error:
```
INFO: Repository details for build/de.nicidienase.chaosflix changed - deleting
INFO: Using git version 2.7.4
INFO: Building version 0.2.8 (23) of de.nicidienase.chaosflix
INFO: Getting source for revision 0.2.8
INFO: Creating local.properties file at build/de.nicidienase.chaosflix/local.properties
INFO: Creating local.properties file at build/de.nicidienase.chaosflix/touch/local.properties
INFO: Cleaning Gradle project...
INFO: Scanning source for common problems...
INFO: Removing gradle-wrapper.jar at gradle/wrapper/gradle-wrapper.jar
INFO: Creating source tarball...
INFO: Building Gradle project...
ERROR: Could not build app de.nicidienase.chaosflix: Build failed for de.nicidienase.chaosflix:0.2.8
==== detail begin ====
NDK is missing a "platforms" directory.
If you are using NDK, verify the ndk.dir is set to a valid NDK directory.  It is currently set to /home/the-scrabi/bin/Android/android-ndk-r10e.
If you are not using NDK, unset the NDK variable from ANDROID_NDK_HOME or local.properties to remove this warning.

Configuration 'compile' in project ':touch' is deprecated. Use 'implementation' instead.
Configuration 'testCompile' in project ':touch' is deprecated. Use 'testImplementation' instead.
touch: 'androidProcessor' dependencies won't be recognized as kapt annotation processors. Please change the configuration name to 'kapt' for these artifacts: 'com.android.databinding:compiler:3.0.1'.
:touch:preBuild UP-TO-DATE
:touch:extractProguardFiles
:touch:preReleaseBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':touch:releaseCompileClasspath'.
> Could not find de.nicidienase.chaosflix:common:1.3-SNAPSHOT.
  Searched in the following locations:
      file:/home/the-scrabi/bin/Android/Sdk/extras/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      file:/home/the-scrabi/bin/Android/Sdk/extras/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      file:/home/the-scrabi/bin/Android/Sdk/extras/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      file:/home/the-scrabi/bin/Android/Sdk/extras/google/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      file:/home/the-scrabi/bin/Android/Sdk/extras/google/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      file:/home/the-scrabi/bin/Android/Sdk/extras/google/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      file:/home/the-scrabi/bin/Android/Sdk/extras/android/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      file:/home/the-scrabi/bin/Android/Sdk/extras/android/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      file:/home/the-scrabi/bin/Android/Sdk/extras/android/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      file:/home/the-scrabi/.m2/repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      file:/home/the-scrabi/.m2/repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      file:/home/the-scrabi/.m2/repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      https://jcenter.bintray.com/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      https://jcenter.bintray.com/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      https://jcenter.bintray.com/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      https://oss.sonatype.org/content/repositories/snapshots/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      https://oss.sonatype.org/content/repositories/snapshots/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      https://oss.sonatype.org/content/repositories/snapshots/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      https://dl.google.com/dl/android/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      https://dl.google.com/dl/android/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      https://dl.google.com/dl/android/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      https://repo1.maven.org/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      https://repo1.maven.org/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      https://repo1.maven.org/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
  Required by:
      project :touch

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
2 actionable tasks: 2 executed
==== detail end ====
INFO: Build for app de.nicidienase.chaosflix failed:
Build failed for de.nicidienase.chaosflix:0.2.8
==== detail begin ====
NDK is missing a "platforms" directory.
If you are using NDK, verify the ndk.dir is set to a valid NDK directory.  It is currently set to /home/the-scrabi/bin/Android/android-ndk-r10e.
If you are not using NDK, unset the NDK variable from ANDROID_NDK_HOME or local.properties to remove this warning.

Configuration 'compile' in project ':touch' is deprecated. Use 'implementation' instead.
Configuration 'testCompile' in project ':touch' is deprecated. Use 'testImplementation' instead.
touch: 'androidProcessor' dependencies won't be recognized as kapt annotation processors. Please change the configuration name to 'kapt' for these artifacts: 'com.android.databinding:compiler:3.0.1'.
:touch:preBuild UP-TO-DATE
:touch:extractProguardFiles
:touch:preReleaseBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':touch:releaseCompileClasspath'.
> Could not find de.nicidienase.chaosflix:common:1.3-SNAPSHOT.
  Searched in the following locations:
      file:/home/the-scrabi/bin/Android/Sdk/extras/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      file:/home/the-scrabi/bin/Android/Sdk/extras/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      file:/home/the-scrabi/bin/Android/Sdk/extras/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      file:/home/the-scrabi/bin/Android/Sdk/extras/google/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      file:/home/the-scrabi/bin/Android/Sdk/extras/google/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      file:/home/the-scrabi/bin/Android/Sdk/extras/google/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      file:/home/the-scrabi/bin/Android/Sdk/extras/android/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      file:/home/the-scrabi/bin/Android/Sdk/extras/android/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      file:/home/the-scrabi/bin/Android/Sdk/extras/android/m2repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      file:/home/the-scrabi/.m2/repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      file:/home/the-scrabi/.m2/repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      file:/home/the-scrabi/.m2/repository/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      https://jcenter.bintray.com/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      https://jcenter.bintray.com/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      https://jcenter.bintray.com/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      https://oss.sonatype.org/content/repositories/snapshots/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      https://oss.sonatype.org/content/repositories/snapshots/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      https://oss.sonatype.org/content/repositories/snapshots/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      https://dl.google.com/dl/android/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      https://dl.google.com/dl/android/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      https://dl.google.com/dl/android/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
      https://repo1.maven.org/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/maven-metadata.xml
      https://repo1.maven.org/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.pom
      https://repo1.maven.org/maven2/de/nicidienase/chaosflix/common/1.3-SNAPSHOT/common-1.3-SNAPSHOT.jar
  Required by:
      project :touch

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
2 actionable tasks: 2 executed
==== detail end ====
INFO: Finished
INFO: 1 build failed
```

**Comment by katrinleinweber** [comment]
Hm :-( I'm not a developer, just an F-Droid user. I'm sorry. It looks like a configuration problem, but apart from `try`ing what the message suggestions and googling the errors, I don't know how to approach this.

**Comment by theScrabi** [comment]
This error is not risen from fdroid. Its comming from gradle, because all that fdroid does is call `gradle assemble`.

**Comment by NiciDieNase** [comment]
The problem is:
> Could not find de.nicidienase.chaosflix:common:1.3-SNAPSHOT.

That's the module that contains everything that is shared between the touch and leanback(TV) apps.
So far this is not published to any repository, so gradle can't find it, if it hasn't been installed locally before building.

**Comment by theScrabi** [comment]
Hm I see. Have you considered to use [jitpack](https://jitpack.io). It will make a repository out of your githup repo.

**Comment by probonopd** [comment]
Unfortunately https://search.f-droid.org/?q=ChaosFlix is not showing anything. Hard to believe ;-/

**Comment by NiciDieNase** [comment]
New features (like chromecast-support) were a higher priority for me than more distribution-channels.
Unfortunately, because chromecast-support requires play services, in it's current state I can't put the app on f-droid.
But since there have been a few requests for f-droid now, it's quite likely that this will be one of the next tasks I'll take on.

**Comment by probonopd** [comment]
Maybe one could make a chaosflix-oss version without chromecast-support then?

**Comment by NiciDieNase** [comment]
That's the plan. To have an additional build-flavor without chromecast/google-play that can be put on f-droid.
Just copying the codebase and removing play services would just be unmanagable in the long run.

In the meantime, you can still get the current build here on github: https://github.com/NiciDieNase/chaosflix/releases

**Comment by genofire** [comment]
Any update?

**Comment by Bubu** [comment]
https://gitlab.com/fdroid/rfp/issues/1075

**Comment by licaon-kter** [comment]
@NiciDieNase How does one build the Free Touch flavour?

This metadata:
```
Builds:
  - versionName: '0.5'
    versionCode: 1
    commit: '0.5'
    subdir: touch
    gradle:
      - free
    rm:
      - leanback
    prebuild:
      - sed -i -e '/appcenter/d' ../common/build.gradle
      - sed -i -e "s/versionfile/\.\.\/versionfile/" build.gradle
      - sed -i -e "s/, ':leanback'//" ../settings.gradle
```

Yields this:
```
DEBUG: Directory: build/de.nicidienase.chaosflix.touch/touch
DEBUG: > /home/strech/fdroidserver-0814/gradlew-fdroid assembleFreeRelease
Found 5.1.1 via distributionUrl
Running /home/strech/.cache/fdroidserver/gradle/5.1.1/bin/gradle assembleFreeRelease

FAILURE: Build failed with an exception.

* What went wrong:
Task 'assembleFreeRelease' not found in project ':touch'. Some candidates are: 'assembleDevFreeRelease'.

* Try:
Run gradle tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
```

But running gradle directly fails with
```
> Task :common:compileDevFreeReleaseKotlin FAILED
e: /home/strech/fdroiddata-0814/build/de.nicidienase.chaosflix.touch/common/src/free/java/de/nicidienase/chaosflix/common/AnalyticsWrapper.kt: (4, 27): Unresolved reference: Application
e: /home/strech/fdroiddata-0814/build/de.nicidienase.chaosflix.touch/common/src/main/java/de/nicidienase/chaosflix/common/viewmodel/PreferencesViewModel.kt: (35, 45): Unresolved reference: startAnalytics
e: /home/strech/fdroiddata-0814/build/de.nicidienase.chaosflix.touch/common/src/main/java/de/nicidienase/chaosflix/common/viewmodel/PreferencesViewModel.kt: (37, 44): Unresolved reference: stopAnalytics
```

**Comment by NiciDieNase** [comment]
The gradle task should be assembleProdFreeRelease.
So propably just replace free with prodFree?

**Comment by licaon-kter** [comment]
Yes, but that fails with the Analytics error as shown above.

**Comment by NiciDieNase** [comment]
The issues above should be solved with the newest commits on develop.
Also the build should work even when the appcenterId properties are not defined, they are not needed for the free build flavour anyways.

**Comment by AreYouLoco** [comment]
I am also waiting to get it from fdroid. Could I help somehow?

**Comment by NiciDieNase** [comment]
Started working on fdroid-metadata
https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6291

**Comment by IzzySoft** [comment]
@NiciDieNase just in case you've only missed the notifications (I know GitLab's sometimes do not arrive): [your MR](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6291) waits on response from you for 5+ months now and will be closed due to inactivity soon unless you "call back there".

**Comment by IzzySoft** [comment]
@NiciDieNase just to let you know that according to our [Issue Cleanup Policy](https://gitlab.com/fdroid/wiki/-/wikis/Internal/Issue-Cleanup-Policy), the merge request now switches from "wait-for-submitter" to "candidate-to-close". If there is no activity on the MR or this issue in the next 4 weeks, we will close the MR to reduce the number of open issues.

**Comment by genofire** [comment]
I am really sad, that no updates of this issue happens round about a year.
